pkg/storage/stores/shipper/compactor/deletion/delete_requests_store.go

```diff
@@ -106,6 +111,10 @@ func (ds *deleteRequestsStore) addDeleteRequest(ctx context.Context, userID stri
 	writeBatch.Add(DeleteRequestsTableName, fmt.Sprintf("%s:%s", deleteRequestDetails, userIDAndRequestID),
 		[]byte(rangeValue), []byte(query))
 
+	// create a gen number for this result
+	writeBatch.Add(DeleteRequestsTableName, fmt.Sprintf("%s:%s", cacheGenNum, userID),
+		[]byte{}, []byte(strconv.FormatInt(time.Now().UnixNano(), 10)))
```

Comment:
do I have this right, the cache gen number is the unix nano timestamp as a base10 integer? is there much of a possibility that the way we generate that value would change? (should there be a `generateCacheGenNum` function?)